### ISSUE TITLE
Hummingbird desktop embedder doesn't build when web dependencies are included

### ISSUE THREAD

**Issue by DaveShuckerow** [flutter/devtools#1122]
This makes sense, but it means that we can't use the linux embedder as a development environment without separating the dart:html and dart:js dependencies into a separate package.

So long as the html and js dependencies are not included in the package, it does properly build.

**Comment by DaveShuckerow** [comment]
This appears to come from the recently added package:sse_client dependency:

```
Compiler message:
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:7:8: Error: Not found: 'dart:html'
import 'dart:html';                                                     
       ^                                                                
../../../../../../.pub-cache/hosted/pub.dartlang.org/http-0.12.0+2/lib/src/browser_client.dart:6:8: Error: Not found: 'dart:html'
import 'dart:html';                                                     
       ^                                                                
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:27:3: Error: Type 'EventSource' not found.
  EventSource _eventSource;                                             
  ^^^^^^^^^^^                                                           
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:46:10: Error: Type 'Event' not found.
  Stream<Event> get onOpen => _eventSource.onOpen;                      
         ^^^^^                                                          
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:68:34: Error: Type 'Event' not found.
  void _onIncomingControlMessage(Event message) {                       
                                 ^^^^^                                  
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:77:27: Error: Type 'Event' not found.
  void _onIncomingMessage(Event message) {                              
                          ^^^^^                                         
../../../../../../.pub-cache/hosted/pub.dartlang.org/http-0.12.0+2/lib/src/browser_client.dart:97:25: Error: Type 'HttpRequest' not
found.
  void _openHttpRequest(HttpRequest request, String method, String url, 
                        ^^^^^^^^^^^                                     
../../../../../../.pub-cache/hosted/pub.dartlang.org/http-0.12.0+2/lib/src/browser_client.dart:32:25: Error: 'HttpRequest' isn't a
type.
  final _xhrs = new Set<HttpRequest>();                                 
                        ^^^^^^^^^^^                                     
Syncing files to device macOS...                                        
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:27:3: Error: 'EventSource' isn't a type.
  EventSource _eventSource;                                                                                        
  ^^^^^^^^^^^                                                                                                      
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:36:9: Error: The method 'EventSource'
isn't defined for the class 'SseClient'.
 - 'SseClient' is from 'package:sse/client/sse_client.dart'                                                        
 ('../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart').
Try correcting the name to the name of an existing method, or defining a method named 'EventSource'.               
        EventSource('$serverUrl?sseClientId=$clientId', withCredentials: true);                                    
        ^^^^^^^^^^^                                                                                                
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:68:34: Error: 'Event' isn't a type.
  void _onIncomingControlMessage(Event message) {                                                                  
                                 ^^^^^                                                                             
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:69:28: Error: 'MessageEvent' isn't a type.
    var data = (message as MessageEvent).data;                                                                     
                           ^^^^^^^^^^^^                                                                            
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:77:27: Error: 'Event' isn't a type.
  void _onIncomingMessage(Event message) {                                                                         
                          ^^^^^                                                                                    
../../../../../../.pub-cache/hosted/pub.dartlang.org/sse-2.1.1/lib/client/sse_client.dart:79:32: Error: 'MessageEvent' isn't a type.
        jsonDecode((message as MessageEvent).data as String) as String;                                            
                               ^^^^^^^^^^^^                                                                        
../../../flutter/packages/flutter/lib/src/painting/_network_image_web.dart:56:12: Error: Method not found:         
'webOnlyInstantiateImageCodecFromUrl'.
    return ui.webOnlyInstantiateImageCodecFromUrl(resolved); // ignore: undefined_function                         
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                     
../../../../../../.pub-cache/hosted/pub.dartlang.org/http-0.12.0+2/lib/src/browser_client.dart:46:19: Error: Method not found:
'HttpRequest'.
    var xhr = new HttpRequest();                                                                                   
                  ^^^^^^^^^^^                                                                                      
../../../../../../.pub-cache/hosted/pub.dartlang.org/http-0.12.0+2/lib/src/browser_client.dart:57:45: Error: Method not found:
'Blob'.
      var blob = xhr.response == null ? new Blob([]) : xhr.response;                                               
                                            ^^^^                                                                   
../../../../../../.pub-cache/hosted/pub.dartlang.org/http-0.12.0+2/lib/src/browser_client.dart:58:24: Error: Method not found:
'FileReader'.
      var reader = new FileReader();                                                                               
                       ^^^^^^^^^^                                                                                  
../../../../../../.pub-cache/hosted/pub.dartlang.org/http-0.12.0+2/lib/src/browser_client.dart:97:25: Error: 'HttpRequest' isn't a
type.
  void _openHttpRequest(HttpRequest request, String method, String url,  
```

Short-term, we can work around this by including sse_client in html_shim and accept that sse won't work in desktop builds.

Long-term, if we want a desktop build of DevTools to be viable, then we'd need to get sse working without dart:html.

**Comment by devoncarew** [comment]
More generally, I think we want to make sure that we're aware of when changes that are committed to devtools break the internal sync of the tool.

@DaveShuckerow - is there a copybara sync set up? @srawlins - would it be useful to set up a per commit/PR test of this project?

**Comment by DaveShuckerow** [comment]
@devoncarew This isn't related to an internal sync of DevTools.  This is related to running DevTools in the Flutter desktop embedder.

There is a copybara script set up, but for right now while we're changing the package structure of devtools, we've introduced changes that break the script.

**Comment by devoncarew** [comment]
> This is related to running DevTools in the Flutter desktop embedder.

Gotcha

**Comment by DaveShuckerow** [comment]
Fixed by #1143
